cas-server-core/src/main/java/org/jasig/cas/util/CompressionUtils.java

```diff
@@ -126,6 +128,22 @@ public static String encodeBase64(final byte[] data) {
     }
 
     /**
+     * Encrypt using the given cipher, and encode the data in base 64.
+     *
+     * @param data the data
+     * @param cipher the cipher
+     * @return the encoded piece of data in base64
+     */
+    public static String encryptAndEncodeBase64(final String data, final Cipher cipher) {
+        try {
+            final byte[] cipherData = cipher.doFinal(data.getBytes());
```

Comment:
I would explicitly declare UTF-8 charset here in coercing the string to bytes. We recommend UTF-8 elsewhere, and it's arguably the safest encoding generally and here in particular.